### ISSUE TITLE
Implement saved pin feature

### ISSUE THREAD

**Issue by jkwening** [TheDevPath/googleMaps-offline-navigator#185]
Currently after adding marker to map, clicking save button generates an alert notifying user of the marker location.

Code snippet from LeafletOsmMap component:
```
    L.DomEvent.on(saveBtn, 'click', function() {
      // TODO - save marker location to db if possible or queue to
      // save when backend server is available
      alert(`Save marker at: ${event.latlng}!`);
    });
```
Refactor code so that it actually saves the given marker to the backend database. Note - accessing saved pins db requires user authentication; if needed feel free to relax that requirement for now. You can do so by commenting out
```router.post('/savedpins', verifyToken, savedPinsController.postSavedPins);```
with note needs to be added back for user authentication and adding a similar code snippet without the 'verifyToken' part.

**Comment by jkwening** [comment]
completed in #185

**Comment by shiolovesgod** [comment]
Hey @jkwening I'm working on #183 (loading saved pins) and I need to be able to save pins to really test the code.  I'll use generated pins until then, but just wanted to know if there was an ETA on this task?

Not sure how far along you are but maybe this snippet will help you:

```
import { makeRequest } from "../../js/server-requests-utils"; 

      let place_id = 'THIS PLACE LABEL';
      
      makeRequest('GET','user')
      .then(res => {
          //user signed in, generate pin data
          let pinData = {
            lat: event.latlng.lat,
            lng: e.latlng.lng,
            place_id: '',
            user: res.data._id
            };
          
          return pinData;

        }
      )
      .then(pinData =>{

        //post saved pin
        makeRequest('POST','savedPin',pinData)
        .then(res=>{
          //pin saved
        })
        .catch(err=>{
          //pin not saved
        });

      })
      .catch(err=>{
        //USER NOT SIGNED IN...
        
        //redirect to sign in page? or overlay signin?

      });
```

**Comment by jkwening** [comment]
@shiolovesgod the save pin feature was merged into development last night (PR #203). If you pull down from upstream development branch into your local development branch and then merge into the branch you're working in for this issue, you'll be all synced up. Thanks.

**Comment by shiolovesgod** [comment]
Great! Checked last night, should have checked again this morning. Thanks.
